router/batchrouter/batchrouter.go

```diff
@@ -2410,3 +2410,24 @@ func (brt *HandleT) updateProcessedEventsMetrics(statusList []*jobsdb.JobStatusT
 		}).Count(count)
 	}
 }
+
+func TryUnmarshalJSON(data []byte, v interface{}) (err error) {
+	c := cap(data)
+	l := len(data)
+
+	startingData := make([]byte, l)
+	copy(startingData, data)
+
+	defer func() {
+		if r := recover(); r != nil {
+			pkgLogger.Warnf("Panic while unmarshalling json: %v", r)
+			pkgLogger.Warnf("	starting slice: [%d:%d]: %q", l, c, string(startingData))
```

Comment:
[optional] Not sure `%q` makes sense here, it'd probably generate confusion at least for me since you got two more double quotes around. Same below.